src/irmin/s.ml

```diff
@@ -558,6 +558,19 @@ module type COMMIT_HISTORY = sig
   (** Same as {{!Private.Node.GRAPH.closure} GRAPH.closure} but for the history
       graph. *)
 
+  val iter :
+    [> `Read ] t ->
+    min:node list ->
+    max:node list ->
+    ?commit:(commit -> unit Lwt.t) ->
+    ?edge:(node -> node -> unit Lwt.t) ->
+    ?skip:(node -> bool Lwt.t) ->
+    ?rev:bool ->
+    unit ->
+    unit Lwt.t
+  (** [iter min max commit edge skip rev ()] iterates over the closure of [t] as
+      specified by {{!Irmin__.S.NODE_GRAPH.iter} GRAPH.iter}. *)
```

Comment:
> `Irmin__.S.NODE_GRAPH.iter`

This looks wrong to me. It shouldn't be necessary to link to hidden modules like that. Perhaps `Irmin.GRAPH` works?

Reply:
I can remove it here, indeed, but there are other places where I reference hidden modules [here](https://github.com/mirage/irmin/pull/1077/files#diff-7ca2203abcf876ec7e28cfe77b5a0b56R376), [here](https://github.com/mirage/irmin/pull/1077/files#diff-ff18801d87f4a3470e94b83d8f20ee06R138) and I can't make them work otherwise.

Reply:
Let's just leave it. We can do a restructuring pass over the documentation later, once Odoc 2 is out.